apps/backoffice-v2/src/common/components/organisms/Filters/Filters.Input.tsx

```diff
@@ -0,0 +1,19 @@
+import { MultiselectFilterInput } from '@/common/components/organisms/Filters/filter-inputs/MultiSelectFilterInput';
+import { IFilterInputParams } from '@/common/components/organisms/Filters/interfaces';
+import { TFilterInputType } from '@/common/components/organisms/Filters/types';
+import { FunctionComponent, useMemo } from 'react';
+
+const getFilterInputComponent = (
+  type: TFilterInputType,
+): React.ComponentType<IFilterInputParams> => {
+  if (type === 'multi-select')
+    return MultiselectFilterInput as React.ComponentType<IFilterInputParams>;
+
+  throw new Error(`Filter input for type ${type} not implemented.`);
+};
+
+export const FiltersInput: FunctionComponent<IFilterInputParams> = props => {
+  const InputComponent = useMemo(() => getFilterInputComponent(props.filter.type), [props.filter]);
+
+  return <InputComponent {...props} />;
+};
```

Comment:
Why do we need this complexity?

Reply:
Generic filters so they could be re-used. What do you mean by complexity?
Each input has its own definition params. Currently just `multi-select`

Reply:
A filter is just an input. This should be something we consider maybe once we have 2 if not 3 different filters. Handling it via composition will probably be easier to maintain.